### PR TITLE
BUG: Fixed Performance Issue In semantic_segmentor.py

### PR DESCRIPTION
**Pull request by vqdang**
Carry on from #285 with a dedicated branch for this fix.
Linked to #287 

Check list:
- [x] Check feature extractor integrity
- [x] Performance gain measurement: 21.67s (new) vs 45.564 (old) using a 4k x 4k WSI

@ByteHexler To add clarification on the origin of this performance issue. 

https://github.com/TissueImageAnalytics/tiatoolbox/blob/ca0ece69e0f1cd110687a391b9aed882597c46d9/tiatoolbox/models/engine/semantic_segmentor.py#L670-L673
`self._process_predictions` is called every time `cum_output` is updated. We will process `n**2` time compared to the normal `n` times. As such, depending on the size of the WSI, this can lead to a significant slowdown, and potentially crash the system because many machines won't be able to hold an entire WSI in the memory. In the initial design, we assumed that

https://github.com/TissueImageAnalytics/tiatoolbox/blob/ca0ece69e0f1cd110687a391b9aed882597c46d9/tiatoolbox/models/engine/semantic_segmentor.py#L681 which calls https://github.com/TissueImageAnalytics/tiatoolbox/blob/ca0ece69e0f1cd110687a391b9aed882597c46d9/tiatoolbox/models/engine/semantic_segmentor.py#L736 with `free_prediction=True` will free up the prediction in place, however, this did not work as expected.

### CODE REVIEW

File: tiatoolbox/models/engine/semantic_segmentor.py [hunk 1]
```diff
@@ -366,26 +366,31 @@ class SemanticSegmentor:
...
+            `docs <https://tia-toolbox.readthedocs.io/en/latest/pretrained.html>`_.
+            By default, the corresponding pretrained weights will also be
+            downloaded. However, you can override with your own set of weights
+            via the `pretrained_weights` argument. Argument is case insensitive.
         pretrained_weights (str): Path to the weight of the corresponding
-          `pretrained_model`.
+            `pretrained_model`.
         batch_size (int) : Number of images fed into the model each time.
-        num_loader_workers (int) : Number of workers to load the data.
+        nu  m_loader_workers (int) : Number of workers to load the data.
```
Comment (shaneahmed) [inline tiatoolbox/models/engine/semantic_segmentor.py:380]:
```suggestion
        num_loader_workers (int) : Number of workers to load the data.
```

**Review by shaneahmed — APPROVED** [review]
Thanks @vqdang
I will merge once @simongraham  has approved it.

**Comment by shaneahmed** [comment]
#287

**Comment by shaneahmed** [comment]
#287

**Review by simongraham — APPROVED** [review]
Looks fine.

**Review by simongraham — APPROVED** [review]

**Review by simongraham — APPROVED** [review]
Looks fine.

**Pull request merged**